services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/events.py

```diff
@@ -216,9 +236,43 @@ async def action(cls, app: FastAPI, scheduler_data: SchedulerData) -> None:
         dynamic_sidecar_client = get_dynamic_sidecar_client(app)
         dynamic_sidecar_endpoint = scheduler_data.dynamic_sidecar.endpoint
 
-        logger.info("Calling into dynamic-sidecar to restore state")
-        await dynamic_sidecar_client.service_restore_state(dynamic_sidecar_endpoint)
-        logger.info("State restored by dynamic-sidecar")
+        # disable file system event watcher while writing
+        # the outputs directory to avoid data being pushed
+        # via nodeports upon change
+        await dynamic_sidecar_client.service_disable_dir_watcher(
+            dynamic_sidecar_endpoint
+        )
+
+        # below tasks can take a while, running them in parallel
+        await logged_gather(
+            dynamic_sidecar_client.service_restore_state(dynamic_sidecar_endpoint),
+            dynamic_sidecar_client.service_pull_output_ports(dynamic_sidecar_endpoint),
+        )
+
+        # inside this directory create the missing dirs, fetch those form the labels
+        director_v0_client: DirectorV0Client = _get_director_v0_client(app)
+        simcore_service_labels: SimcoreServiceLabels = (
```

Comment:
so is this where you need the labels of the image. With this you created a dependency of the dynamic sidecar towards the director-v0 for getting the image labels. Although, you have a docker engine at hand, you could directly get the labels instead. Otherwise you could pass them down from the director-v2.
Anyway, this is not the purpose of this PR.